src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private Object[] array;
+    private int counter = 0;
+
+    public ArrayList() {
+        array = new Object[ARRAY_SIZE];
+    }
+
+    private void resize(int minSize) {
+        int newSize = minSize + (minSize >> 1);
+        if (newSize <= 0) {
+            throw new ArrayIndexOutOfBoundsException("Array is out of integer bounds and "
+                    + "require hugeCapacity() implementation. It's can't be bigger right now!");
+        }
+        Object[] oldArray = array;
+        array = new Object[newSize];
+        System.arraycopy(oldArray, 0, array,0, counter);
+    }
 
     @Override
     public void add(T value) {
-
+        if (counter >= array.length - 1) {
+            resize(counter);
+        }
+        array[counter] = value;
+        counter = counter + 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (counter >= array.length - 1) {
+            resize(counter);
+        }
+        System.arraycopy(array, index, array, index + 1, counter - index);
+        array[index] = value;
+        counter = counter + 1;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > array.length) {
+            resize(list.size());
+        }
```

Comment:
this is redundant - you already have capacity check in your add() method that you use in a loop